pkg/exporter/process.go

```diff
@@ -616,26 +634,35 @@ func (ep *ExportingProcess) dataRecSanityCheck(rec entities.Record) error {
 }
 
 func createClientConfig(config *ExporterTLSClientConfig) (*tls.Config, error) {
-	roots := x509.NewCertPool()
-	ok := roots.AppendCertsFromPEM(config.CAData)
-	if !ok {
-		return nil, fmt.Errorf("failed to parse root certificate")
-	}
-	if config.CertData == nil {
-		return &tls.Config{
-			RootCAs:    roots,
-			MinVersion: tls.VersionTLS12,
-			ServerName: config.ServerName,
-		}, nil
-	}
-	cert, err := tls.X509KeyPair(config.CertData, config.KeyData)
-	if err != nil {
-		return nil, err
-	}
-	return &tls.Config{
-		Certificates: []tls.Certificate{cert},
-		RootCAs:      roots,
-		MinVersion:   tls.VersionTLS12,
+	tlsMinVersion := config.MinVersion
+	// This should already be the default value for tls.Config, but we duplicate the earlier
+	// implementation, which was explicitly setting it to 1.2.
+	if tlsMinVersion == 0 {
+		tlsMinVersion = tls.VersionTLS12
+	}
+	// #nosec G402: client is in charge of setting the min TLS version. We use 1.2 as the
+	// default, which is secure.
+	tlsConfig := &tls.Config{
 		ServerName:   config.ServerName,
-	}, nil
+		CipherSuites: config.CipherSuites,
+		MinVersion:   tlsMinVersion,
+	}
+	// Use system roots if config.CAData == nil.
+	if config.CAData != nil {
+		roots := x509.NewCertPool()
+		ok := roots.AppendCertsFromPEM(config.CAData)
+		if !ok {
+			return nil, fmt.Errorf("failed to parse root certificate")
+		}
+		tlsConfig.RootCAs = roots
+	}
+	// Don't use a client certificate if config.CertData == nil.
+	if config.CertData != nil {
```

Comment:
Do we need to check KeyData != nil as well?

Reply:
I saw you do this check for the collector. Maybe we should do the same way for consistency? But I don't have strong opinion on that.